packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Wysiwyg/Field.tsx

```diff
@@ -51,6 +62,89 @@ const Wysiwyg = React.forwardRef<EditorApi, WysiwygProps>(
       setMediaLibVisible(false);
     };
 
+    const handlePasteImage = async (files: File[]) => {
+      if (isUploading || files.length === 0) return;
+
+      setIsUploading(true);
+
+      try {
+        // Create FormData for the upload
+        const formData = new FormData();
```

Comment:
I think this whole block of logic about uploading is too low level for this package. What would be a more viable solution would be to access the useUpload hook that the upload plugin uses internally. Unfortunately it's not currently exported, but you could set that up:
- `import { useUpload } from '@strapi/upload/strapi-admin'` here
- add an `exports.ts` file to `upload/admin/src` that re-exports useUpload, and would allow us to expose more in the future. Just like it's done in the content-manager
- `export * from './exports'` in `upload/admin/src/index`

Reply:
I agree what the situation is not ideal, but I think what you have is fine for now.

About how we can properly fix it in the future. I think the best approach could be to have an API that allows to pass a promise to the notification API, so it can handle loading/success/error all in one (just like in sonner, see [the 'promise' button there](https://sonner.emilkowal.ski/))